### ISSUE TITLE
CookieAuthenticationHandler for Identity.Application scheme fires twice

### ISSUE THREAD

**Issue by ypedegroot** [DuendeSoftware/Support#1072]
**Which version of Duende IdentityServer are you using?**
`7.0.0-rc.2`

**Which version of .NET are you using?**
`.NET 8`

**Describe the bug**
When requesting a page, the log entry `"AuthenticationScheme: Identity.Application was not authenticated."` is always produced twice.

**To Reproduce**
Clone your example v7/5_AspNetIdentity, run the IdentityServerAspNetIdentity project and check the log.

**Expected behavior**
The log entry `"AuthenticationScheme: Identity.Application was not authenticated."` is shown only once.

**Log output with stacktrace**
```
[14:59:19 Information] Serilog.AspNetCore.RequestLoggingMiddleware
HTTP GET / responded 200 in 2.2686 ms

[14:59:19 Information] Serilog.AspNetCore.RequestLoggingMiddleware
HTTP GET /favicon.ico responded 304 in 0.1142 ms

[14:59:21 Debug] Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationHandler
AuthenticationScheme: Identity.Application was not authenticated.

[14:59:21 Debug] Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationHandler
AuthenticationScheme: Identity.Application was not authenticated.

[14:59:21 Information] Serilog.AspNetCore.RequestLoggingMiddleware
HTTP GET / responded 200 in 1.8100 ms
```

**Additional context**
When using the Quickstart, I noticed that you use IdentityServer version 6.2.0. However, the problem is still the same there.

**Comment by AndersAbel** [comment]
`app.UseIdentityServer()` includes a call to `app.UseAuthentication()`. Do you have another explicit call to `app.UseIdentityServer` as well? That would explain why it runs twice.

Running the authentication middleware twice looks bad in the logs. But there is as far as we know not a problem and will not break any thing.

**Comment by josephdecock** [comment]
The "Is not authenticated" log message is written any time the AuthenticateAsync method is called with an anonymous request. Multiple authenticate async calls on a single incoming request is actually normal. 

AuthenticateAsync determines who are what sent the http request, and produces a result that indicates either an anonymous request, or else a authentication ticket, which describes the identity of the requestor (the claims principal) as well as session metadata (the authentication properties). 

AuthenticateAsync is called automatically in the authentication middleware, and the resulting claims principal within the ticket is stored in the HttpContext.User. However, the authentication properties within the ticket are not stored anywhere. So, any time that either ASP.NET, ASP.NET Identity, IdentityServer, or UI code needs authentication properties, there will be an additional call to the AuthenticateAsync method. The authentication handler is implemented to expect this, and only does the actual work of reading/deserializing/unprotecting the auth ticket only once per request.

There isn't anything specific to IdentityServer that is causing additional AuthenticateAsync calls. For example, try doing

`dotnet new webapp --auth Individual -n IndividualAuth`

And change the log level for Microsoft.AspNetCore.Authentication to debug. I get 7 different Identity.Application not authenticated messages logged by the framework on a single request.

In short, this is a debug level log message from ASP.NET itself, and isn't anything to worry about.

Finally, your Additional debugging note is correct. We create a cookie auth scheme for use as the default cookie, and we create an external authentication scheme that is used to manage external authentication. The fact that 2 schemes are registered doesn't have any bearing on this log message though, because this log message is specific to the one default scheme that is trying to authentication.
